pkg/kv/kvserver/replica_raftstorage.go

```diff
@@ -792,7 +804,16 @@ func (r *Replica) clearSubsumedReplicaDiskData(
 ) error {
 	// NB: we don't clear RangeID local key spans here. That happens
 	// via the call to preDestroyRaftMuLocked.
-	getKeySpans := rditer.MakeReplicatedKeySpansExceptRangeID
+	getKeySpans := func(d *roachpb.RangeDescriptor) []roachpb.Span {
+		return rditer.Select(d.RangeID, rditer.SelectionOptions{
+			StateMachineSelectionOptions: rditer.StateMachineSelectionOptions{
+				ReplicatedBySpan: roachpb.RSpan{
+					Key:    d.StartKey,
+					EndKey: d.EndKey,
```

Comment:
Is there any use-case where `StartKey`/`EndKey` is not taken from the range descriptor? If so, directly passing the descriptor (instead of taking range ID + span manually) can be considered "safer".

Reply:
It feels wrong to use a `RangeDescriptor` here. That is a very large struct. But I was able to replace this with just `d.RSpan()`.